playwright/src/test/java/com/microsoft/playwright/TestBrowserContextViewport.java

```diff
@@ -0,0 +1,120 @@
+package com.microsoft.playwright;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+
+import static com.microsoft.playwright.Utils.verifyViewport;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class TestBrowserContextViewport extends TestBase {
+
+  @Test
+  void shouldGetTheProperDefaultViewPortSize() {
+    verifyViewport(page, 1280, 720);
+  }
+
+  @Test
+  void shouldSetTheProperViewportSize() {
+    verifyViewport(page, 1280, 720);
+    page.setViewportSize(123, 456);
+    verifyViewport(page,123, 456);
+  }
+
+  @Test
+  void shouldReturnCorrectOuterWidthAndOuterHeight() {
+    LinkedHashMap size = (LinkedHashMap) page.evaluate("() => {\n" +
+      "  return {\n" +
+      "    innerWidth: window.innerWidth,\n" +
+      "    innerHeight: window.innerHeight,\n" +
+      "    outerWidth: window.outerWidth,\n" +
+      "    outerHeight: window.outerHeight,\n" +
+      "  };\n" +
+      "}");
+    assertEquals(1280, size.get("innerWidth"));
+    assertEquals(720, size.get("innerHeight"));
+    assertTrue((int)size.get("outerWidth") >= (int)size.get("innerWidth"));
+    assertTrue((int)size.get("outerHeight") >= (int)size.get("innerHeight"));
+  }
+
+  @Test
+  void shouldEmulateDeviceWidth() {
+    verifyViewport(page, 1280, 720);
+    page.setViewportSize(200, 200);
+    assertEquals(200, page.evaluate("() => window.screen.width"));
+    assertTrue((Boolean) page.evaluate("() => matchMedia('(min-device-width: 100px)').matches"));
+    assertFalse((Boolean) page.evaluate("() => matchMedia('(min-device-width: 300px)').matches"));
+    assertFalse((Boolean) page.evaluate("() => matchMedia('(max-device-width: 100px)').matches"));
+    assertTrue((Boolean) page.evaluate("() => matchMedia('(max-device-width: 300px)').matches"));
+    assertFalse((Boolean) page.evaluate("() => matchMedia('(device-width: 500px)').matches"));
+    assertTrue((Boolean) page.evaluate("() => matchMedia('(device-width: 200px)').matches"));
+    page.setViewportSize(500, 500);
+    assertEquals(500, page.evaluate("() => window.screen.width"));
+    assertTrue((Boolean) page.evaluate("() => matchMedia('(min-device-width: 400px)').matches"));
```

Comment:
nit: you can avoid explicit cast with `assertEquals(true, page...` but it's up to you.